sympy/stats/stochastic_process_types.py

```diff
@@ -971,37 +950,372 @@ def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
         Probability of the condition.
 
         """
-        new_condition, new_givencondition = self._rvindexed_subs(condition, given_condition)
 
-        if isinstance(new_givencondition, RandomSymbol):
-            condrv = random_symbols(new_condition)
-            if len(condrv) == 1 and condrv[0] == new_givencondition:
-                return BernoulliDistribution(self.probability(new_condition), 0, 1)
-
-            if any([dependent(rv, new_givencondition) for rv in condrv]):
-                return Probability(new_condition, new_givencondition)
-            else:
-                return self.probability(new_condition)
-
-        if new_givencondition is not None and \
-                not isinstance(new_givencondition, (Relational, Boolean)):
-            raise ValueError("%s is not a relational or combination of relationals"
-                    % (new_givencondition))
-        if new_givencondition == False:
-            return S.Zero
-        if new_condition == True:
-            return S.One
-        if new_condition == False:
-            return S.Zero
-        if not isinstance(new_condition, (Relational, Boolean)):
-            raise ValueError("%s is not a relational or combination of relationals"
-                    % (new_condition))
-        if new_givencondition is not None:  # If there is a condition
-        # Recompute on new conditional expr
-            return self.probability(given(new_condition, new_givencondition, **kwargs), **kwargs)
-        return pspace(new_condition).probability(new_condition, **kwargs)
+        return _probability_stoch('Bernoulli', condition, given_condition, evaluate, **kwargs)
 
     def density(self, x):
         return Piecewise((self.p, Eq(x, self.success)),
                          (1 - self.p, Eq(x, self.failure)),
                          (S.Zero, True))
+
+
+dict_rv_subs = {'Poisson': lambda rv: Poisson(rv.name, lamda=rv.pspace.process.lamda*rv.key),
+                'Bernoulli': lambda rv: Bernoulli(rv.name, p=rv.pspace.process.p,
+                            succ=rv.pspace.process.success, fail=rv.pspace.process.failure)
```

Comment:
Instead of dict, may be, RVs can be added to their related to process as a property or attribute.